### ISSUE TITLE
`ReprCString` should have drop, like `CVec`

### ISSUE THREAD

**Issue by kenfred** [h33p/cglue#14]
If `ReprCString` is analogous to Rust's `String` and owns its own memory, it needs a drop method just like `CVec`. As it is currently, passing `ReprCString` to the C side will leak because it has no way to free the string.

**Comment by h33p** [comment]
This is true, thanks for reporting! Generally it's a bit tricky, because it's meant to kind of mimic regular C `char *` strings, so I'm not set whether it's a good idea to add a drop with each string, or make sure each string is allocated with `malloc`.
